### DISCUSSION TITLE
Expanding width of a channel as it moves to another stage?

### DISCUSSION THREAD

**Discussion by AlanKrueger** [nxt3AT/sankeydiagram.net#61] [Ideas]
Most Sankey diagrams track populations of things that are fairly immutable, like currency units and people, things like can't be added to or subtracted from without incoming or outgoing channels on the graph to explain them.

What happens, though, when you have a population that could multiply at different stages? How might that be represented in one of these diagrams, and would that be possible to accomplish with this tool?

For instances, say you're tracking six-packs of soda and you have something like this:

    Input [100] First Stage
    First Stage [80] Second Stage
    First Stage [20] Trash
    Second Stage [480] Third Stage

Clearly this doesn't line up, but suppose the Second Stage splits the six-packs into individual cans. The amount in equals the amount out, but there's a different unit involved.

I could probably make this work if there was a way to have separate input and output counts for a given line, something like this:

    Input [100] First Stage
    First Stage [80] Second Stage
    First Stage [20] Trash
    Second Stage [80 -> 480] Third Stage

This last line indicates that 80 of the things from the second stage result in 480 items reaching the third stage. The resulting graph could be sized on one side to 80 and on the other side to 480, with labels on both sides to indicate the amounts.

Thoughts?

**Comment by JonasDoesThings** [comment]
Hi, thx for the suggestion.
I see what you mean, but since it's a rather niche use-case imo and would require quite some development, I am currently not planning on implementing this due to time constraints on my side.

I would be open to good implementations via a Pull Request tho.

**Comment by AlanKrueger** [comment]
Looking at the source, it looks like you're using https://github.com/ricklupton/d3-sankey-diagram as the underlying diagramming library. I'm assuming I'd probably need to raise this issue there.

**Reply by JonasDoesThings** [reply]
Yep, that library is used! It's modified and hacked-into, but I would probably be able to backport a change from upstream there :)
